src/Microsoft.Android.Sdk.Analysis/CodeFixes/CustomApplicationCodeFixProvider.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+[ExportCodeFixProvider (LanguageNames.CSharp, Name = nameof (CustomApplicationCodeFixProvider)), Shared]
+public class CustomApplicationCodeFixProvider : CodeFixProvider
+{
+	private const string title = "Fix Activation Constructor";
+	public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create (CustomApplicationAnalyzer.DiagnosticId);
+
+	public sealed override FixAllProvider GetFixAllProvider ()
+	{
+		return WellKnownFixAllProviders.BatchFixer;
+	}
+
+	public override async Task RegisterCodeFixesAsync (CodeFixContext context)
+	{
+		var root = await context.Document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
+		var diagnostic = context.Diagnostics.First ();
+		var diagnosticSpan = diagnostic.Location.SourceSpan;
+		var classDeclaration = root.FindToken (diagnosticSpan.Start).Parent.AncestorsAndSelf ()
+				.OfType<ClassDeclarationSyntax> ().First ();
+		context.RegisterCodeFix (CodeAction.Create (title, c =>
+			InjectConstructorAsync (context.Document, classDeclaration, c), equivalenceKey: title), diagnostic);
+	}
+
+	private async Task<Document> InjectConstructorAsync (Document document, ClassDeclarationSyntax classDeclaration, CancellationToken cancellationToken)
+	{
+		var constructor = CreateConstructorWithParameters (classDeclaration.Identifier);
+		var newClassDeclaration = classDeclaration.AddMembers (constructor);
+		var root = await document.GetSyntaxRootAsync (cancellationToken);
+		var newRoot = root.ReplaceNode (classDeclaration, newClassDeclaration);
+		return document.WithSyntaxRoot (newRoot);
+	}
+
+	private ConstructorDeclarationSyntax CreateConstructorWithParameters (SyntaxToken identifier)
+	{
+		var parameters = SyntaxFactory.ParameterList (SyntaxFactory.SeparatedList (new [] {
+				SyntaxFactory.Parameter (SyntaxFactory.Identifier ("javaReference"))
+					.WithType (SyntaxFactory.ParseTypeName ("IntPtr")),
+				SyntaxFactory.Parameter (SyntaxFactory.Identifier ("transfer"))
+					.WithType (SyntaxFactory.ParseTypeName ("JniHandleOwnership"))
```

Comment:
If you test this code fix in an IDE (could even be VS Code), does it auto-add the namespace/using at the top?

Reply:
not sure. I'm having difficulty getting this to run in the IDE

Reply:
I decided to make it easy on myself and just put in the fully qualified type string.

Reply:
Sometimes testing these, I found the IDE just put the `using` for me at the top. But I never figured out what was doing it.

It was this codefix:

https://github.com/jonathanpeppers/memory-analyzers/blob/2bc83bcb6ab0fd4673a646f1b7bbb2e574cdeac5/MemoryAnalyzers/MemoryAnalyzers.CodeFixes/MemoryAnalyzersCodeFixProvider.cs#L122

It somehow puts `using System.Diagnostics.CodeAnalysis;` at the top when I tested this.